CDCgov/prime-simplereport
# Reduce parameters in OrganizationMutationResolver Part 1/3

# BACKEND PULL REQUEST

## Related Issue

- Partially implements #5682

## Changes Proposed

- Remove unused mutations createOrganization, addFacilityNew, updateFacilityNew from schema files, mutation resolver and unit testing.
- Create a new input type for the mutations: addFacility, updateFacility.
- Add conditional logic in the mutation resolver to also read the facility data from the new input types.
- Regenerate graphql file.

## Additional Information
- Low test coverage will not be addressed in this PR as the code not covered will be removed in the last part of this effort.
- A common type ProviderInput was created.
- The input types for the mutations were moved to their own file.

## Testing
Code is available in dev3
- Verify that add a facility and update facility in the settings page works as usual.
- Verify that the mutations addFacility and updateFacility work as expected when the facility information is passed through the new input types. Use mutations below to test this with postman.

```
mutation AddFacility(
  $testingFacilityName: String!
  $cliaNumber: String
  $street: String!
  $streetTwo: String
  $city: String
  $state: String!
  $zipCode: String!
  $phone: String
  $email: String
  $orderingProviderFirstName: String
  $orderingProviderMiddleName: String
  $orderingProviderLastName: String
  $orderingProviderSuffix: String
  $orderingProviderNPI: String
  $orderingProviderStreet: String
  $orderingProviderStreetTwo: String
  $orderingProviderCity: String
  $orderingProviderState: String
  $orderingProviderZipCode: String
  $orderingProviderPhone: String
  $devices: [ID]!
) {
  addFacility(
    facilityInfo:{
    facilityName: $testingFacilityName
    cliaNumber: $cliaNumber
    address:{
      street: $street
      streetTwo: $streetTwo
      city: $city
      state: $state
      zipCode: $zipCode
    }
    phone: $phone
    email: $email
    orderingProvider:{
      firstName: $orderingProviderFirstName
      middleName: $orderingProviderMiddleName
      lastName: $orderingProviderLastName
      suffix: $orderingProviderSuffix
      npi: $orderingProviderNPI
      street: $orderingProviderStreet
      streetTwo: $orderingProviderStreetTwo
      city: $orderingProviderCity
      state: $orderingProviderState
      zipCode: $orderingProviderZipCode
      phone: $orderingProviderPhone
    }
    deviceIds: $devices
    }
  ) {
    id
    __typename
  }
}

{
  "testingFacilityName":"Hufflepuff",
  "cliaNumber":"12D4567891",
  "street":"123 main street",
  "streetTwo":"",
  "city":"",
  "state":"NJ",
  "zipCode":"07601",
  "phone":"2012789876",
  "email":"hufflepuff@hogwarts.edu",
  "orderingProviderFirstName":"Minerva",
  "orderingProviderMiddleName":"",
  "orderingProviderLastName":"McGonagall",
  "orderingProviderSuffix":"",
  "orderingProviderNPI":"1234567890",
  "orderingProviderStreet":"",
  "orderingProviderStreetTwo":"",
  "orderingProviderCity":"",
  "orderingProviderState":"",
  "orderingProviderZipCode":"",
  "orderingProviderPhone":"2711872345",
  "devices":["61456e84-fc0c-42c1-9e82-360105364aad"]
}

mutation UpdateFacility(
  $facilityId: ID!
  $testingFacilityName: String!
  $cliaNumber: String
  $street: String!
  $streetTwo: String
  $city: String
  $state: String!
  $zipCode: String!
  $phone: String
  $email: String
  $orderingProviderFirstName: String
  $orderingProviderMiddleName: String
  $orderingProviderLastName: String
  $orderingProviderSuffix: String
  $orderingProviderNPI: String
  $orderingProviderStreet: String
  $orderingProviderStreetTwo: String
  $orderingProviderCity: String
  $orderingProviderState: String
  $orderingProviderZipCode: String
  $orderingProviderPhone: String
  $devices: [ID]!
) {
  updateFacility(
    facilityInfo: {
      facilityId: $facilityId
      facilityName: $testingFacilityName
      cliaNumber: $cliaNumber
      address:{
        street: $street
        streetTwo: $streetTwo
        city: $city
        state: $state
        zipCode: $zipCode
      }
      phone: $phone
      email: $email
      orderingProvider: {
        firstName: $orderingProviderFirstName
        middleName: $orderingProviderMiddleName
        lastName: $orderingProviderLastName
        suffix: $orderingProviderSuffix
        npi: $orderingProviderNPI
        street: $orderingProviderStreet
        streetTwo: $orderingProviderStreetTwo
        city: $orderingProviderCity
        state: $orderingProviderState
        zipCode: $orderingProviderZipCode
        phone: $orderingProviderPhone
      }
      deviceIds: $devices
    }
  ) {
    id
    __typename
  }
}

{
  "facilityId":"c085c1fc-ef23-45de-856a-fa3a57ddcb17",
  "testingFacilityName":"Hufflepuff",
  "cliaNumber":"12D4567891",
  "street":"123 main st",
  "streetTwo":"",
  "city":"",
  "state":"NJ",
  "zipCode":"07601",
  "phone":"2012789876",
  "email":"hufflepuff@hogwarts.edu",
  "orderingProviderFirstName":"Pomona",
  "orderingProviderMiddleName":"",
  "orderingProviderLastName":"Sprout",
  "orderingProviderSuffix":"",
  "orderingProviderNPI":"1234567890",
  "orderingProviderStreet":"",
  "orderingProviderStreetTwo":"",
  "orderingProviderCity":"",
  "orderingProviderState":"",
  "orderingProviderZipCode":"",
  "orderingProviderPhone":"2711872345",
  "devices":["61456e84-fc0c-42c1-9e82-360105364aad"]
}


```
